_includes/archive-single.html

```diff
@@ -11,7 +11,7 @@
 {% endif %}
 
 <div class="{{ include.type | default: 'list' }}__item">
-  <article class="archive__item" itemscope itemtype="https://schema.org/CreativeWork">
+  <article class="archive__item" itemscope itemtype="https://schema.org/CreativeWork"{% if page.locale %} lang="{{ post.locale | default: site.locale | replace: "_", "-" | default: "en" }}"{% endif %}>
```

Comment:
With `{% if %}` guarding the presence of the per-post locale setting, all the `default`s can be safely dropped.

Specific to this one: Did you mix up `page` and `post`?

Reply:
I had the exact same thought five seconds after pushing ;-) I'll update the PR.

Reply:
Fixed In b5fd4dae